src/shogun/kernel/ShiftInvariantKernel.cpp

```diff
@@ -34,12 +34,12 @@
 
 using namespace shogun;
 
-ShiftInvariantKernel::ShiftInvariantKernel() : Kernel(0)
+ShiftInvariantKernel::ShiftInvariantKernel() : RandomMixin<Kernel>(0)
 {
 	register_params();
 }
 
-ShiftInvariantKernel::ShiftInvariantKernel(const std::shared_ptr<Features >&l, const std::shared_ptr<Features >&r) : Kernel(l, r, 0)
+ShiftInvariantKernel::ShiftInvariantKernel(const std::shared_ptr<Features >&l, const std::shared_ptr<Features >&r) : RandomMixin<Kernel>(l, r, 0)
```

Comment:
this is not needed here, can be done in the gk class itself i think

Reply:
Sure, but then there's the pain of deriving from RandomMixin on **every** ShiftInvariant kernel derivative. Or we could just use the power of inheritance.

Reply:
but this mixin is not used in every subclass (as we will only implement the spectral density for a few kernels)... so this just causes a lot of code being unnecessarily compiled.

Reply:
There's a design challenge though,

I want this preprocessor to be flexible and not just stick to Gaussian Kernel. Now that ShiftInvariantKernel is stripped of its RandomMixin, we would be using m_prng from the Gaussian class. If we continue by this design, we'd need to add all their header files.